library/src/scala/annotation/MacroAnnotation.scala

```diff
@@ -31,14 +31,20 @@ trait MacroAnnotation extends StaticAnnotation:
    *    def transform(using Quotes)(tree: quotes.reflect.Definition): List[quotes.reflect.Definition] =
    *      import quotes.reflect._
    *      tree match
-   *        case DefDef(name, TermParamClause(param :: Nil) :: Nil, tpt, Some(rhsTree)) =>
-   *          (Ref(param.symbol).asExpr, rhsTree.asExpr) match
```

Comment:
Why are all these macros here and in run-macros changed? Does the old implementation not work anymore?

Reply:
In this case, the old implementation still works but is suboptimal and has a bug. It had three issues:
* With quotes getting unpickled in with the wrong owner each splice has the potential to trigger an owner change. 
* If users would assemble the expression using reflection they would need extra explicit `changeOwner` calls
* To recover the type of the signature we should look at the signature and not the terms. In the previous version, we had an accidental term reference to the parameter of the annotated def that leaked into the new field.

Reply:
An alternative way to construct the RHSs will be added in the documentation in #16454. 

https://github.com/lampepfl/dotty/pull/16454/files#diff-0a9170b85c5b55edde099043317e7f1e48b3d8998b34707be1596b8161e40f89R85-R165

Reply:
To make the context changes simpler, I plan to add a method like `inContext`.

Something like
```scala
val rhs = inQuotes(owner){ '{...} }